uaclient/log.py

```diff
@@ -0,0 +1,59 @@
+import json
+import logging
+import re
+from collections import OrderedDict
+from typing import Any, Dict, List
+
+
+class JsonArraryFormatter(logging.Formatter):
+    default_time_format = "%Y-%m-%dT%H:%M:%S%z"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not self.datefmt:
+            self.datefmt = self.default_time_format
+        self._required_fields = self.get_required_fields()
+
+    def get_required_fields(self) -> List[str]:
+        # We only support the % style formatting...currently
+        if isinstance(self._style, logging.PercentStyle):
+            formatter_style_pattern = re.compile(r"%\((.+?)\)", re.IGNORECASE)
+        else:
+            raise ValueError("Invalid format: %s" % self._fmt)
+
+        if self._fmt:
+            return formatter_style_pattern.findall(self._fmt)
+        else:
+            return []
```

Comment:
This would be useful if we were writing this as a library, but we only need it to work for our specific use case.

I suggest we just hard-code our list of fields somewhere in this file rather than parsing them from the _fmt string.

Reply:
Okay, I had this question...should it be generic or custom made to fit our use case...thanks for answering.